pkg/resource/deploy/snapshot.go

```diff
@@ -280,3 +292,33 @@ func (snap *Snapshot) withUpdatedResources(update func(*resource.State) *resourc
 	newSnap.Resources = new
 	return &newSnap
 }
+
+type snapshotIntegrityError struct {
+	// The underlying error that caused this integrity error, if applicable.
+	err error
+}
+
+// Creates a new snapshot integrity error with a message produced by the given
+// format string and arguments. Supports wrapping errors with %w. Snapshot
+// integrity errors are raised by Snapshot.VerifyIntegrity when a problem is
+// detected with a snapshot (e.g. missing or out-of-order dependencies, or
+// unparseable data).
+func SnapshotIntegrityErrorf(format string, args ...interface{}) error {
+	return &snapshotIntegrityError{err: fmt.Errorf(format, args...)}
```

Comment:
Would it make helpful to grab `runtime.Stack()` and stick it on the error, and print it out in `printSnapshotIntegrityError` ?

Reply:
Nice suggestion -- done :+1: :smile: